### PR TITLE
[Behat] Maintaining cart after authorization scenarios

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Related tickets |
| License         | MIT

This is a port from old to new Behat scenarios of ``cart_maintenance.feature``.

### CODE REVIEW

File: features/cart/shopping_cart/maintaining_cart_after_authorization.feature [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        Given I have product "Stark T-Shirt" in the cart
+        When I log in as "robb@stark.com" with "KingInTheNorth" password
+        And I see the summary of my cart
+        Then there should be one item in my cart
+        And this item should have name "Stark T-shirt"
+
+    @ui
+    Scenario: Having cart maintained after registration
+        Given I have product "Stark T-Shirt" in the cart
+        When I register with email "eddard@stak.com" and password "handOfTheKing"
```
Comment (pamil) [inline features/cart/shopping_cart/maintaining_cart_after_authorization.feature]: stak.com

Comment (Zales0123) [inline features/cart/shopping_cart/maintaining_cart_after_authorization.feature]: He was working in http://www.stakceramics.com/, it's his business email.

Comment (pamil) [inline features/cart/shopping_cart/maintaining_cart_after_authorization.feature]: These shady dealings will get him buried!

File: features/cart/shopping_cart/maintaining_cart_after_authorization.feature [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    Background:
+        Given the store operates on a single channel in "France"
+        And the store has a product "Stark T-shirt" priced at "€12.00"
+        And there is user "robb@stark.com" identified by "KingInTheNorth"
+
+    @ui
+    Scenario: Having cart maintained after logging in
+        Given I have product "Stark T-Shirt" in the cart
+        When I log in as "robb@stark.com" with "KingInTheNorth" password
+        And I see the summary of my cart
```
Comment (michalmarcinkowski) [inline features/cart/shopping_cart/maintaining_cart_after_authorization.feature]: We should avoid using `see` IMO we can change it to simply `And I view my cart`

@TheMadeleine pick one: 
1. I view my cart
2. I browse my cart
3. I look into my cart 😆

**Comment by michalmarcinkowski** [comment]
No PHP code at all! 🎉🎉🎉

**Comment by michalmarcinkowski** [comment]
Thanks Mateusz! 👍

File: features/cart/shopping_cart/maintaining_cart_after_authorization.feature (hunk 1 above)
Comment (lchrusciel) [inline features/cart/shopping_cart/maintaining_cart_after_authorization.feature]: He has lost his head for this business.

**Pull request merged**